presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -93,6 +97,59 @@ public JoinNode(@JsonProperty("id") PlanNodeId id,
         checkArgument(!isCrossJoin() || inputSymbols.equals(outputSymbols), "Cross join does not support output symbols pruning or reordering");
     }
 
+    public JoinNode flip()
+    {
+        return new JoinNode(
+                getId(),
+                flipType(type),
+                right,
+                left,
+                flipJoinCriteria(criteria),
+                flipOutputSymbols(getOutputSymbols(), left, right),
+                filter,
+                rightHashSymbol,
+                leftHashSymbol,
+                distributionType);
```

Comment:
Okay, so it's not possible to "flip" distribution. What about check-stating there is no distribution in `flip()` and insted
```
JoinNode repartitionedJoin = joinNode.withDistributionType(PARTITIONED);
possibleJoinNodes.add(repartitionedJoin);
possibleJoinNodes.add(repartitionedJoin.flip());
```

having

```
possibleJoinNodes.add(joinNode.withDistributionType(PARTITIONED));
possibleJoinNodes.add(joinNode.flip().withDistributionType(PARTITIONED));
```
?